### PR TITLE
Improve utility zookeeper-adjust-block-numbers-to-parts.

### PR DESCRIPTION
**Pull request by vitlibar**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Improve the utility zookeeper-adjust-block-numbers-to-parts:
- Add filters on shards and tables.
- Add flag --dry-run.
- The utility now checks cversion before doing anything.

Category (leave one):
- Build/Testing/Packaging Improvement

### CODE REVIEW

File: utils/zookeeper-adjust-block-numbers-to-parts/main.cpp [hunk 1]
```diff
@@ -28,36 +70,74 @@ size_t getMaxBlockSizeForPartition(zkutil::ZooKeeper & zk,
...
+
+size_t getCurrentBlockNumberForPartition(zkutil::ZooKeeper & zk, const std::string & part_path)
+{
+    Coordination::Stat stat;
+    zk.get(part_path, &stat);
+
+    /// References:
+    /// https://stackoverflow.com/a/10347910
+    /// https://bowenli86.github.io/2016/07/07/distributed%20system/zookeeper/How-does-ZooKeeper-s-persistent-sequential-id-work/
+    return (stat.cversion + stat.numChildren) / 2;
```
Comment (alesapin) [inline utils/zookeeper-adjust-block-numbers-to-parts/main.cpp]: What if the user uses an older version of zookeeper? Or zookeeper unexpectedly change this rule?

Comment (alesapin) [reply utils/zookeeper-adjust-block-numbers-to-parts/main.cpp]: Maybe better to get a sequential node and parse number from its name?

File: utils/zookeeper-adjust-block-numbers-to-parts/main.cpp [hunk 2]
```diff
@@ -66,67 +146,98 @@ std::unordered_map<std::string, size_t> getAllTablesBlockPaths(zkutil::ZooKeeper
...
+    if (part_paths_with_max_block_numbers.empty())
     {
-        std::cerr << "Rotating on: " << path << std::endl;
-        rotateNodes(zookeeper, path, max_block_num);
+        std::cout << "No adjusting required." << std::endl;
+        return 0;
     }
+
+    /// Adjust the block numbers.
+    if (options.count("dry-run"))
```
Comment (alesapin) [inline utils/zookeeper-adjust-block-numbers-to-parts/main.cpp]: print something about it.

Comment (vitlibar) [reply utils/zookeeper-adjust-block-numbers-to-parts/main.cpp]: done

File: utils/zookeeper-adjust-block-numbers-to-parts/main.cpp (hunk 1 above)
Comment (vitlibar) [reply utils/zookeeper-adjust-block-numbers-to-parts/main.cpp]: > Maybe better to get a sequential node and parse number from its name?

done

**Pull request merged**